store/sqlstore/store.go

```diff
@@ -1450,39 +1438,53 @@ func (ss *SqlStore) migrate(direction migrationDirection) error {
 		}
 	}
 
-	source := bindata.Resource(assetNamesForDriver, func(name string) ([]byte, error) {
-		return migrations.Asset(filepath.Join(ss.DriverName(), name))
+	src, err := mbindata.WithInstance(&mbindata.AssetSource{
+		Names: assetNamesForDriver,
+		AssetFunc: func(name string) ([]byte, error) {
+			return migrations.Asset(filepath.Join(ss.DriverName(), name))
+		},
 	})
-
-	sourceDriver, err := bindata.WithInstance(source)
 	if err != nil {
 		return err
 	}
+	defer src.Close()
+
+	db := ss.GetMasterX().DB.DB
+	if db == nil {
+		return fmt.Errorf("db instance is nil")
+	}
```

Comment:
Do we expect this to be nil, or it's just defence in depth?

Reply:
Maybe it was be too defensive, it's actually going to exit app if there is a problem with opening the db. Can be removed.